include/ck/tensor_operation/gpu/device/device_batched_gemm_softmax_gemm.hpp

```diff
@@ -24,7 +24,8 @@ template <typename ALayout,
           typename B0ElementwiseOperation,
           typename Acc0ElementwiseOperation,
           typename B1ElementwiseOperation,
-          typename CElementwiseOperation>
+          typename CElementwiseOperation,
+          bool MaskOutUpperTriangle> // TODO: enum for mask type
```

Comment:
Is it a bool or a ```MaskingSpecialization```?

Reply:
I intentionally let it stay as it is because this device op (without permute) is probably not going to be used in practice. I think the effort is better spent in grouped and batched permute variants.

Reply:
agreed.